pkg/mounter/oss/ossfs.go

```diff
@@ -96,10 +61,78 @@ func NewFuseOssfs(configmap *corev1.ConfigMap, m metadata.MetadataProvider) Fuse
 }
 
 func (f *fuseOssfs) Name() string {
-	return "ossfs"
+	return OssFsType
 }
 
-func (f *fuseOssfs) PodTemplateSpec(c *FusePodContext, target string) (*corev1.PodTemplateSpec, error) {
+func (f *fuseOssfs) PrecheckAuthConfig(o *Options, onNode bool) error {
+
+	if o.AuthType != AuthTypeRRSA && o.AssumeRoleArn != "" {
+		return fmt.Errorf("only support access OSS through STS AssumeRole when authType is RRSA")
+	}
+
+	switch o.AuthType {
+	case AuthTypePublic:
+	case AuthTypeSTS:
+		// rolename may retrieve from metadata service
+		if onNode && o.RoleName == "" {
+			return fmt.Errorf("missing roleName or ramRole in volume attributes")
+		}
+	case AuthTypeRRSA:
+		if err := checkRRSAParams(o); err != nil {
+			return err
+		}
+	case AuthTypeCSS:
+		if o.SecretProviderClass == "" {
+			return fmt.Errorf("use CsiSecretStore but secretProviderClass is empty")
+		}
+	default:
+		if features.FunctionalMutableFeatureGate.Enabled(features.RundCSIProtocol3) {
```

Comment:
I don't think we need to check protocol in here, @tydra-wang please double check